packages/langchain/src/openai/util.ts

```diff
@@ -40,7 +44,9 @@ function mapToolToOpenAiFunction(tool: StructuredTool): {
   return {
     name: tool.name,
     description: tool.description,
-    parameters: zodToJsonSchema(tool.schema)
+    parameters: isZodSchema(tool.schema)
+      ? zodToJsonSchema(tool.schema)
+      : tool.schema
```

Comment:
So the previous logic (and I keep it unchanged for now as well) is that only if user utilizes LangChain's `tool()` function, we apply `zodToJsonSchema()` to `schema` field if needed. Because `schema` per LangChain's definition can now be `JSONSchema` or Zod schema.

Otherwise, if user directly defines `tools` with our `AzureOpenAiChatCompletionTool` (which means they don't need mapping), they should call `zodToJsonSchema()` if needed.